modules/openapi-generator/src/main/resources/Java/libraries/resttemplate/ApiClient.mustache

```diff
@@ -769,23 +770,32 @@ public class ApiClient{{#jsr310}} extends JavaTimeFormatter{{/jsr310}} {
         ResponseEntity<T> responseEntity = null;
         int attempts = 0;
         while (attempts < maxAttemptsForRetry) {
-            try {
-                responseEntity = restTemplate.exchange(requestEntity, returnType);
-                break;
-            } catch (HttpServerErrorException ex) {
-                attempts++;
-                if (attempts < maxAttemptsForRetry) {
-                    try {
-                        Thread.sleep(waitTimeMillis);
-                    } catch (InterruptedException e) {
-                        Thread.currentThread().interrupt();
-                    }
+          try {
+            responseEntity = restTemplate.exchange(requestEntity, returnType);
+            break;
+          } catch (HttpServerErrorException | HttpClientErrorException ex) {
+            if (ex instanceof HttpServerErrorException
+                || ((HttpClientErrorException) ex)
+                    .getStatusCode()
+                    .equals(HttpStatus.TOO_MANY_REQUESTS)) {
+              attempts++;
+              if (attempts < maxAttemptsForRetry) {
+                try {
+                  Thread.sleep(waitTimeMillis);
+                } catch (InterruptedException e) {
+                  Thread.currentThread().interrupt();
                 }
+              } else {
+                throw ex;
+              }
+            } else {
+              throw ex;
             }
+          }
         }
 
-        if (responseEntity == null) {
-            throw new RestClientException("API returned HttpServerErrorException");
+        if(responseEntity == null) {
+        throw new RestClientException("ResponseEntity is null");
```

Comment:
can you be indent this line properly? thanks

Reply:
have done. thank you. and sorry should done it earlier. 

on related note, are you aware of any efforts to introduce automated code formatting in mustache files (per corresponding language style guide). wanted to check with you before I spend time on it.